src/components/completeExercise/directives/completeExerciseExercise/controllers/completeExerciseBaseZnkExercise.controller.js

```diff
@@ -131,7 +131,9 @@
                 znkSessionDataSrv.isActiveLiveSession().then(function (liveSessionData) {
                     if (exerciseResult.exerciseTypeId !== ExerciseTypeEnum.LECTURE.enum) {
                         var liveSessionOn = !angular.equals(liveSessionData, {});
-                        if (angular.isUndefined(exerciseResult.isReviewed) && liveSessionOn) {
+                        if (ENV.testToTakeExamId && ENV.testToTakeExamId !==  null) {
```

Comment:
You did not check it the current exercise is the which test to take section, you only checked if there's a variable called "ENV.testToTakeExamId".
This implementation means ALL exercises will be marked as "YES".